2-mandatory/3-stocks.js

```diff
@@ -64,10 +79,33 @@ function getPriceChanges(closingPricesForAllStocks) {
     The price should be shown with exactly 2 decimal places.
 */
 function highestPriceDescriptions(closingPricesForAllStocks, stocks) {
-    // TODO
-}
-
-
+    let highestArray = [];
+    let g = 0;
+    for(elementS of closingPricesForAllStocks){
+        let highestPF = 0;
+        let highestP = elementS[0];
+        let highestN = STOCKS[g].toUpperCase();
+        for(i = 0; i <= elementS.length; i++){
+            if (highestP < elementS[i]){
+                highestP = elementS[i];
+            }else highestP = highestP;
+        }
+        highestPF = Number(highestP.toFixed(2));
+        highestArray.push(`The highest price of ${highestN} in the last 5 days was ${highestPF}`);
+        g++;
+        
+    };
+    return highestArray;
+};
+/*
+const CLOSING_PRICES_LAST_5_DAYS_FOR_ALL_STOCKS = [
+    [179.19, 180.33, 176.28, 175.64, 172.99], // AAPL
+    [340.69, 342.45, 334.69, 333.20, 327.29], // MSFT
+    [3384.44, 3393.39, 3421.37, 3420.74, 3408.34], // AMZN
+    [2951.88, 2958.13, 2938.33, 2928.30, 2869.45], // GOOGL
+    [1101.30, 1093.94, 1067.00, 1008.87, 938.53] // TSLA
+];
+*/
```

Comment:
This looks fine, the only issue is that based on the test the price should be shown with exactly 2 decimal places. To fix this, you can simply use `Math.round()`  for heighestP variable